components/loggingInterop.ts

```diff
@@ -62,16 +54,13 @@ export enum LogLevel {
 
 /**
  * Represents the different internal log categories used by Peacock.
+ * @internal
  */
-export enum LogCategory {
+export const enum LogCategory {
     /**
      * Remove the category from the log
      */
     NONE = "none",
-    /**
-     * Set the category to the name of the calling function
-     */
-    CALLER = "caller",
```

Comment:
Drive-by removal of this because it causes a deoptimization to occur for every call of log, meaning any function that calls log will be much slower as a result. For things like recursive or repeated calls, this causes exponential time loss.

fyi @LennardF1989 - future migration will be to add more manual category names. Eventually, we should probably make this required.